### PR TITLE
feat: Create Hero Pattern w/ Macros

### PR DESCRIPTION
**Pull request by jmuzina**
## Done

This is a feature PR to merge the following split PRs that were used to build and document the Hero HOC/Pattern:

- [x] #5189 
- [x] #5229 
- [x] #5230 
- [x] #5231 
- [x] #5242 

Fixes [WD-12212](https://warthogs.atlassian.net/browse/WD-12212)

## QA

- Review [hero docs](https://vanilla-framework-5248.demos.haus/docs/patterns/hero)
- Compare all examples against [design](https://www.figma.com/design/Hll6ZRvHyTTGBZZSoRd8ei/Higher-order-components-(HOCs)?node-id=689-7744&t=bdSNSuP448nO3iel-0)
- Review [4.15 release notes](https://vanilla-framework-5248.demos.haus/docs/whats-new)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots

image

[WD-12212]: https://warthogs.atlassian.net/browse/WD-12212?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jmuzina** [comment]
Blocked pending merge of #5242

**Comment by lyubomir-popov** [comment]
looks awesome!

File: releases.yml [hunk 1]
```diff
@@ -16,6 +16,10 @@
       url: /docs/patterns/navigation#full-width-dropdowns
       status: New
       notes: We've introduced a new full-width dropdown option to the top navigation component.
+    - component: Hero
+      url: /docs/patterns/hero#layouts
+      status: New
+      notes: We've added new layout options to the hero component.
```
Comment (bartaz) [inline releases.yml:22]:
```suggestion
      notes: We've added new layout options and detailed documentation to the hero pattern.
```

File: templates/docs/patterns/hero/index.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+---
+
+A hero is a prominent banner section typically used to quickly capture the user's attention after they land on the page.
+Depending on the size and composition of your content, you can choose from a variety of hero layouts:
+
+- [50/50](#5050)
+- [50/50 with full-width image](#5050-with-full-width-image)
+- [25/75 Signpost](#2575-signpost)
+- [75/25](#7525)
+- [Fallback](#fallback) (default)
```
Comment (bartaz) [inline templates/docs/patterns/hero/index.md:14]: Feels bit weird do call "fallback" a default. What does it even mean. If it's a default, why do we call it "fallback" and document last?

To my understanding this is the "last resort" kind of hero: if content (long title) doesn't fit into any of the standard hero layouts.


```suggestion
- [Fallback](#fallback)
```

**Review by bartaz — DISMISSED** [review]
Couple small suggestions (that I'm gonna commit), and looking good thanks!

**Comment by bartaz** [comment]
Oh well, because I made some edits myself, I can no longer approve and merge it.

I guess it needs to be approved by you now @pastelcyborg

**Pull request merged**